stumpy/stump.py

```diff
@@ -190,6 +190,25 @@ def _compute_diagonal(
                 if T_B_subseq_isconstant[i + k] and T_A_subseq_isconstant[i]:
                     pearson = 1.0
 
+                # due to imprecision, the value of pearson may be outside
+                # of interval [-1, 1]
+                if pearson >= 1.0:
+                    pearson = 1.0
+                if pearson <= -1.0:
+                    pearson = -1.0
+
+                if pearson != 1.0 and pearson >= config.STUMPY_CORRELATION_THRESHOLD:
+                    # refine pearson only when we have to
+                    if pearson > ρ[thread_idx, i, 0] or (
+                        ignore_trivial and np.any(pearson > ρ[thread_idx, i + k])
```

Comment:
Why do we compare `pearson` with all values in `ρ[thread_idx, i + k]` rather than against one single value?

Reply:
I thought it might be possible that `pearson` do not update a matrix profile, but update just the right (or left) matrix profile. In such case, I wanted to use a more-accurate value of `pearson`. However, I now think that might be too much. Because, we mostly care about the precision of the main matrix profile. So, now that you mentioned it, I think it should be okay to just compare it to the `ρ[thread_idx, i + k, 0]`, i.e. the pearson that corresponds to the main matrix profile (not left / right)

Reply:
Cool. I hope this reduces the compute time